it/src/main/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManager.java

```diff
@@ -0,0 +1,390 @@
+
+    private String testId;
+    private String projectId;
+    private String instanceId;
+    private CredentialsProvider credentialsProvider;
+
+    private BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler;
+
+    private Builder(String testId, String projectId) {
+
+      this.testId = testId;
+      this.projectId = projectId;
+      this.instanceId = generateInstanceId(testId);
+    }
+
+    public Builder setTestId(String testId) {
+      this.testId = testId;
+      return this;
+    }
+
+    public String getTestId() {
+      return testId;
+    }
+
+    public Builder setProjectId(String projectId) {
+      this.projectId = projectId;
+      return this;
+    }
+
+    public String getProjectId() {
+      return projectId;
+    }
+
+    public Builder setInstanceId(String instanceId) {
+      this.instanceId = instanceId;
+      return this;
+    }
+
+    public String getInstanceId() {
+      return instanceId;
+    }
+
+    public Builder setCredentialsProvider(CredentialsProvider credentialsProvider) {
+      this.credentialsProvider = credentialsProvider;
+      return this;
+    }
+
+    public CredentialsProvider getCredentialsProvider() {
+      return credentialsProvider;
+    }
+
+    public Builder setBigtableResourceManagerClientHandler(
+        BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler) {
+      this.bigtableResourceManagerClientHandler = bigtableResourceManagerClientHandler;
+      return this;
+    }
+
+    public BigtableResourceManagerClientHandler getBigtableResourceManagerClientHandler() {
+      return bigtableResourceManagerClientHandler;
+    }
+
+    public DefaultBigtableResourceManager build() throws IOException {
+
+      // create the default bigtable resource manager client handler if one was not already set.
+      if (this.bigtableResourceManagerClientHandler == null) {
+
+        // create the bigtable admin and data client settings builders, and set the necessary id's
+        // for each.
+        BigtableInstanceAdminSettings.Builder bigtableInstanceAdminSettings =
+            BigtableInstanceAdminSettings.newBuilder().setProjectId(projectId);
+        BigtableTableAdminSettings.Builder bigtableTableAdminSettings =
+            BigtableTableAdminSettings.newBuilder()
+                .setProjectId(projectId)
+                .setInstanceId(instanceId);
+        BigtableDataSettings.Builder bigtableDataSettings =
+            BigtableDataSettings.newBuilder().setProjectId(projectId).setInstanceId(instanceId);
+
+        // add the credentials to the builders, if set.
+        if (this.credentialsProvider != null) {
+          bigtableInstanceAdminSettings.setCredentialsProvider(this.credentialsProvider);
+          bigtableTableAdminSettings.setCredentialsProvider(this.credentialsProvider);
+          bigtableDataSettings.setCredentialsProvider(this.credentialsProvider);
+        }
+
+        this.bigtableResourceManagerClientHandler =
```

Comment:
I'd avoid altering the builder state in the `build()` method (i.e. create a temporary local bigtableResourceManagerClientHandler variable here instead of updating the instance variable), because this is error-prone. For example, if you do this:
```
builder = DefaultBigtableResourceManager.builder(...);
resourceManager1 = builder.build();
resourceManager2 = builder.setCredentialsProvider(...).build();
```
`resourceManager2` actually won't use the credentials provider. But if you re-create `bigtableResourceManagerClientHandler` on every `build()` call it will.

Reply:
Fixed.